src/reconcile/rec_visibility.c

```diff
@@ -69,6 +69,14 @@ __rec_append_orig_value(
 
     /* Review the current update list, checking conditions that mean no work is needed. */
     for (;; upd = upd->next) {
+        if (upd->txnid != WT_TXN_ABORTED)
```

Comment:
I didn't clearly understand from the code changes that moving the check to the top of the loop is fixing something? Or is it just an optimization?

Reply:
We never check aborted update for

```
/*
         * Prepared updates should already be in the update list, add the original update to the	         * Prepared updates should already be in the update list, add the original update to the
         * list only when the prepared update is a tombstone.	         * list only when the prepared update is a tombstone.
         */	         */
        if (unpack->tw.prepare && upd->type != WT_UPDATE_TOMBSTONE)	
```
It's possible we have an aborted tombstone.

```
        if (unpack->tw.start_ts == upd->start_ts && unpack->tw.start_txn == upd->txnid &&
          upd->type != WT_UPDATE_TOMBSTONE)
```
is less a concern as won't have aborted updates on the disk.

Reply:
Also the code becomes cleaner and we don't need to worry about aborted cases when we add new check.